Server/src/services/db/log.controller.ts

```diff
@@ -64,35 +53,34 @@ export default class LogController {
     }
 
     /**
-     * Updates a log in the database
-     *
-     * @param timestamp - Time of log which should be updated. (Key Pair with trackerId)
-     * @param trackerId - Tracker.uid of Log which should be updated. (Key Pair with timestamp)
-     * @param position - New position after change (optional)
-     * @param heading - New heading after change (optional)
-     * @param speed - New speed after change (optional)
-     * @param battery - New battery after change (optional)
-     * @param data - new optional addtional data field.
+     * Updates an Log entry.
+     * 
+     * @param uid - Indicator for specific log.
+     * @param timestamp - Time when the log was created.
+     * @param position - gps position of the tracker/app.
+     * @param heading - degree in which the tracker/app was pointing.
+     * @param speed - current speed of tracker/app.
+     * @param battery - current battery of tracker.
+     * @param data - GPS Data.
+     * @param vehicleId - which vehicle is connected with said tracker/app. For tracker this can be found in the tracker model.
+     * @param trackerId - identifier for said tracker. For app data this field is always `null`
      * @returns Log | null if an error occurs.
      */
-    public async update(timestamp : Date, trackerId: string, position?: JSON, heading?: number, speed?: number, battery?: number, data?: JSON) : Promise<Log | null> {
+    public async update(uid : number, timestamp? : Date, position? : JSON, heading? : number, speed?: number, battery? : number, data? : JSON, vehicleId? : number, trackerId? : string): Promise<Log | null> {
         try {
-            // TODO: vvv This.
-            let pos = JSON.parse(JSON.stringify(position)) as Prisma.InputJsonObject
-            let d = (data === undefined ? Prisma.JsonNull : JSON.parse(JSON.stringify(data))) as Prisma.InputJsonObject
             return await this.prisma.log.update({
                 where: {
-                    timestamp_trackerId: {
-                        timestamp: timestamp,
-                        trackerId: trackerId
-                    }
+                    uid : uid
                 },
                 data : {
-                    position: pos,
+                    timestamp : timestamp,
+                    position: (position as unknown as Prisma.InputJsonValue),
                     heading: heading,
                     speed: speed,
                     battery: battery,
-                    data: d
+                    data: (data as unknown as Prisma.InputJsonValue),
+                    vehicleId : vehicleId,
+                    trackerId : trackerId

```

Comment:
You are inconsistent with your colon placements, we always like to put it with no preceding but a following white space character (e.g. `timestamp: timestamp`).

Same applies for types (e.g. `uid: number`)